mslib/mswms/dataaccess.py

```diff
@@ -224,6 +238,25 @@ def _determine_filename(self, variable, vartype, init_time, valid_time, reload=T
                               variable, vartype, init_time, valid_time, type(ex), ex)
                 raise ValueError(f"variable type {vartype} not available for variable {variable}")
 
+    def is_reload_required(self, filenames):
+        reload = False
+        for filename in filenames:
+            basename = os.path.basename(filename)
+            if basename not in self._file_cache:
+                reload = True
+                break
+            fullname = os.path.join(self._root_path, basename)
+            if not os.path.exists(fullname):
+                reload = True
+                break
+            mtime = os.path.getmtime(fullname)
```

Comment:
Because we change files and while we can do this incorrectly we could try to catch a problem here and log it.